play-json/shared/src/test/scala/play/api/libs/json/MacroSpec.scala

```diff
@@ -327,6 +327,12 @@ class MacroSpec extends WordSpec with MustMatchers
       )
     }
 
+    "handle case class with default values, format defined in companion object" in {
+      val json = Json.obj("id" -> 15)
+      val expected = WithDefaultInCompanion(15, "a")
+      Json.fromJson[WithDefaultInCompanion](json).get mustEqual expected
+    }
```

Comment:
Is this related to the `pure` change? If it's just a new test, that's OK, but I want to make sure I understand it if it's somehow testing something to do with `pure`.

Reply:
That's the tests failing at #78 due to the invalid laziness

Reply:
@richdougherty updated